lyse/__init__.py

```diff
@@ -553,31 +654,27 @@ def save_result(self, name, value, group=None, overwrite=True):
             PermissionError: A `PermissionError` is raised if an attribute with
                 name `name` already exists but `overwrite` is set to `False`.
         """
-        if self.no_write:
-            msg = "Cannot save result; this instance is read-only."
-            raise PermissionError(msg)
-        with h5py.File(self.h5_path,'a') as h5_file:
-            if not group:
-                if self.group is None:
-                    msg = """Cannot save result; no default group set. Either
-                        specify a value for this method's optional group
-                        argument, or set a default value using the set_group()
-                        method."""
-                    raise ValueError(dedent(msg))
-                # Save to analysis results group by default
-                group = 'results/' + self.group
-            elif not group in h5_file:
-                # Create the group if it doesn't exist
-                h5_file.create_group(group) 
-            if name in h5_file[group].attrs and not overwrite:
-                msg = """Cannot save result; group '{group}' already has
-                    attribute '{name}' and overwrite is set to False. Set
-                    overwrite=True to overwrite the existing value.""".format(
-                        group=group,
-                        name=name,
-                    )
-                raise PermissionError(dedent(msg))
-            set_attributes(h5_file[group], {name: value})
+        if not group:
+            if self.group is None:
+                msg = """Cannot save result; no default group set. Either
+                    specify a value for this method's optional group
+                    argument, or set a default value using the set_group()
+                    method."""
+                raise ValueError(dedent(msg))
+            # Save to analysis results group by default
+            group = 'results/' + self.group
+        elif not group in self.h5_file:
```

Comment:
Unrelated to this PR, but can you change this to `group not in self.h5_file` while you are here? It is equivalent, but more readable syntax

Reply:
Looks like there are a couple of others too in some later methods

Reply:
I think I have found them all. There were quite a few.